pkg/client/daemon/outbound.go

```diff
@@ -135,18 +137,40 @@ func (o *outbound) routerServerWorker(c context.Context) (err error) {
 // in the search path declared in the Docker config. The "tel2-search" domain fills this purpose and a request for
 // "<single label name>.tel2-search." will be resolved as "<single label name>." using the search path of this resolver.
 const tel2SubDomain = "tel2-search"
-const dotTel2SubDomain = "." + tel2SubDomain
-const dotKubernetesZone = "." + kubernetesZone
+const dotTel2SubDomain = "." + tel2SubDomain + "."
+const dotKubernetesZone = "." + kubernetesZone + "."
 
 var localhostIPv6 = []net.IP{{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}}
 var localhostIPv4 = []net.IP{{127, 0, 0, 1}}
 
+func (o *outbound) doClusterLookup(query string) bool {
+	if strings.HasSuffix(query, dotKubernetesZone) && strings.Count(query, ".") < 4 {
+		return false
+	}
+
+	query = query[:len(query)-1] // skip last dot
+
+	// Always include configured includeSuffixes
+	for _, sfx := range o.dnsConfig.IncludeSuffixes {
+		if strings.HasSuffix(query, sfx) {
+			return true
+		}
+	}
+
+	// Skip configured excludeSuffixes
+	for _, sfx := range o.dnsConfig.ExcludeSuffixes {
+		if strings.HasSuffix(query, sfx) {
+			return false
+		}
+	}
+	return true
+}
+
 func (o *outbound) resolveInCluster(c context.Context, qType uint16, query string) []net.IP {
-	query = query[:len(query)-1]
 	query = strings.ToLower(query) // strip of trailing dot
```

Comment:
I think that `// strip of trailing dot` had been intended for the previous deleted line, and therefore should also be deleted?

Reply:
Yup. That comment should not be there.